src/Orckestra.Composer.Cart/Providers/Order/EditingOrderProvider.cs

```diff
@@ -53,6 +56,110 @@ public virtual async Task<bool> CanEdit(Overture.ServiceModel.Orders.Order order
             return isOrderEditable;
         }
 
+        public virtual async Task<bool> CanCancel(Overture.ServiceModel.Orders.Order order)
```

Comment:
as discussed please rename to have  and calculate inside both values
GetCancellationStatus()

CancellationStatus { bool CanCancel {get; set; }; bool CancellationPending {get; set; }; }

to make code readable, you can extract separate methods for example to get messages or others